VMware: vmware_dvs_portgroup - Add PVLAN support

OnceUponALoop opened the pull request:
##### SUMMARY
Adds support for pvlan assignment to portgroups.

Fixes: #47649

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
vmware_dvs_portgroup

##### ADDITIONAL INFORMATION
I added the flag `vlan_pvlan` to mimic the existing `vlan_trunk` flag.

A better fix would be to create a `vlan_type` flag and allow it to be either `trunk` or `pvlan` but that would break existing configuration and I didn't want to make that decision

Commit:
Add PVLAN support - fixes #47649

Commit:
Fix linting and documentation bugs

Commit:
fix version_added to match latest

mariolenz commented:
Thanks for this contribution! At first glance it looks good to me. Just two things:

1) You should check if `vlan_pvlan` differs from the current configuration which is done in the method `check_dvspg_state`. This will allow people to change from a non-private VLAN to a private one and vice versa.
2) I think you can't have a private VLAN trunk so you should probably define `vlan_pvlan` and `vlan_trunk` as `mutually_exclusive`. Just grep for this in the VMware modules and you should find an example how this can be done.

Oh, and I guess the ansible team will want to have a [changelog fragment](https://docs.ansible.com/ansible/2.9/community/development_process.html#changelogs-how-to). I think something like this would be a `minor_changes`.

Commit:
make vlan_pvlan and vlan_trunk exclusive
update check_dvspg_state to support pvlan

Commit:
fixed linting

Commit:
added changelog fragment

OnceUponALoop commented:
@mariolenz 
Thanks for the assist - I rushed the PR and should've considered both suggestions - good to go now, added them as suggested, added the changelog fragment, and tested.

Commit:
fix incorrect implementation after testing

goneri reviewed (changes requested):
Thanks @OnceUponALoop for the contribution. Could you extend the functional test to cover the new feature? The test playbook is here: https://github.com/ansible/ansible/blob/devel/test/integration/targets/vmware_dvs_portgroup/tasks/main.yml

Thanks :-)

Akasurde commented:
@OnceUponALoop Could you please add a test case for this change? Thanks.

needs_info

mariolenz commented:
@OnceUponALoop maybe you can join forces with @axarriola who issued the similar PR #67205.
